Scope optional limits

EgleHelms opened the pull request:
@davehorton @kitajchuk This was the code that was lost somehow during the scoping. 
As agreed, we wanted the new call limits under the env VITE_APP_ENABLE_ACCOUNT_LIMITS_ALL to be disabled for non-admin users, and the MAX limits hidden completely.

Account User view with env VITE_APP_ENABLE_ACCOUNT_LIMITS_ALL
![image](https://user-images.githubusercontent.com/63853378/210963558-0a12bda2-8154-4f8a-9083-289cbba3376d.png)

Admin user view env VITE_APP_ENABLE_ACCOUNT_LIMITS_ALL
![image](https://user-images.githubusercontent.com/63853378/210963389-0350a758-93cd-4bee-8861-6e479c5fc6a0.png)


There is still a question on where we should handle the deletion of the limits set the first time that are in contradiction with the new limits set (e.g. we set session limits, then we decide to change to minutes limits the session limits are not deleted all limits we send more limits and all exist in the backend - so the old ones should be deleted). @davehorton @kitajchuk what do you think should we do this in the frontend?

davehorton commented:
> There is still a question on where we should handle the deletion of the limits set the first time that are in contradiction with the new limits set (e.g. we set session limits, then we decide to change to minutes limits the session limits are not deleted all limits we send more limits and all exist in the backend - so the old ones should be deleted). @davehorton @kitajchuk what do you think should we do this in the frontend?

Yes, I think this should be done in the front end (explicitly deleting the "old" limits)

EgleHelms commented:
@kitajchuk @davehorton this is ready for review.

Changes:
- Service provider Limits disabled when the env is enabled
- No limits or uinits are available to change for non-admin users (all are disabled)
- Max limits both minute or session are hidden for service_provider and account users
- quantity is set to "" when unit is changed to initiate deletion

davehorton commented:
@EgleHelms  Perhaps I am doing something wrong but in testing this I am seeing an error in the console and I am seeing neither the Units nor the Licensed minutes on the Account screen.  I am logged in as admin user and my .env.local has `VITE_APP_ENABLE_ACCOUNT_LIMITS_ALL=true`

![image](https://user-images.githubusercontent.com/1884973/212703895-ae06d93c-4999-46e7-b3e4-0f02e65e28d7.png)

kitajchuk commented:
> @EgleHelms Perhaps I am doing something wrong but in testing this I am seeing an error in the console and I am seeing neither the Units nor the Licensed minutes on the Account screen. I am logged in as admin user and my .env.local has `VITE_APP_ENABLE_ACCOUNT_LIMITS_ALL=true`
> 
> ![image](https://user-images.githubusercontent.com/1884973/212703895-ae06d93c-4999-46e7-b3e4-0f02e65e28d7.png)

That’s a react key error—should be easily fixable but I’m curious why it would show up suddenly 🤔

davehorton commented:
let me retest and make sure I am not doing anything stupid

davehorton commented:
yeah I retested to make sure I am testing this PR and it does look like a bug here

davehorton commented:
I put in some logging and I think the issue is that "category" is used as the key and it is not unique
```
filtered limits: [{"label":"Max calls","category":"voice_call_session"},{"label":"Licensed calls","category":"voice_call_session_license"}]
```

EgleHelms commented:
hmm, I wonder, why would it not be unique? the four categories have different category names. When I was fixing, it was not popping up.
@kitajchuk would you just add a random id to the key to create a unique one? something like key={`${category}-${Math.floor + Math.random() * 100000 )}` It seems to solve the problem, but I am not sure if this is the right solution or is this a dumb solution?

Also, this error, was not preventing me from viewing the fields. They also work to submit the limits too.

![image](https://user-images.githubusercontent.com/63853378/212887893-fc7f470e-f0c3-4dbc-bf4a-b40d7889eb66.png)

EgleHelms commented:
@kitajchuk @davehorton  sorry this is a bit dumb issue I left in, the issue was that I have added another Component <ScopedAccess> that one needs to have a key too since it is also within the map function. So I added the key and the issue is gone. Though the error in the console did not prevent me from viewing the fields and it and somehow is not always there in the console, so I have missed it.

kitajchuk commented:
> @kitajchuk @davehorton sorry this is a bit dumb issue I left in, the issue was that I have added another Component `<ScopedAccess>` that one needs to have a key too since it is also within the map function. So I added the key and the issue is gone. Though the error in the console did not prevent me from viewing the fields and it and somehow is not always there in the console, so I have missed it.

This code got merged with duplicate keys explicitly coded into the map—this is not good. Can we please fix that. Should be easy—even just using string template. Or remove the key on the fragment as it looks like that is not needed—only the direct component returned at the root of the map callback needs a key. 

https://reactjs.org/docs/lists-and-keys.html#keys

kitajchuk commented:
> My bad for merging that, @EgleHelms maybe just commit this change on the open PR for date fix (which also needs to be rebased to main)?

Just remove the fragment as it’s no longer needed since you added the ScopedAccess wrapper.
